src/components/structures/RoomView.js

```diff
@@ -74,6 +75,8 @@ module.exports = React.createClass({
             syncState: MatrixClientPeg.get().getSyncState(),
             hasUnsentMessages: this._hasUnsentMessages(room),
             callState: null,
+            readReceiptEventId: room.getEventReadUpTo(MatrixClientPeg.get().credentials.userId),
+            readMarkerGhostEventId: undefined,
```

Comment:
okay, i've read this about 3 times and I still don't get what the difference are between readReceipt and readMarker, and what a ghost really is.  This feels like as good a place as any to just explain the algorithm here in the context of what the IDs are actually describing...